test/test_quantization.py

```diff
@@ -793,6 +813,31 @@ def checkQuantized(model):
         model = quantize_qat(model, test_only_train_fn, self.img_data)
         checkQuantized(model)
 
+    def test_train_save_load_eval(self):
+        qengine = 'fbgemm'
+        model = ManualLinearQATModel()
+        model = prepare_qat(model)
+        self.checkObservers(model)
+        test_only_train_fn(model, self.train_data)
+        model = convert(model)
+
+        model = quantize_qat(ManualLinearQATModel(), test_only_train_fn,
+                             self.train_data)
+        quant_state_dict = model.state_dict()
+
+        x = torch.rand(2, 5, dtype=torch.float)
+        ref = model(x)
+
+        # Check eval using quantized state_dict
+        model = ManualLinearQATModel()
+
+        model.qconfig = torch.quantization.get_default_qat_qconfig(qengine)
+        torch.quantization.prepare_qat(model, inplace=True)
+        torch.quantization.convert(model, inplace=True)
+        model.load_state_dict(quant_state_dict)
+
+        out = model(x)
+        self.assertEqual(ref, out)
```

Comment:
How about testing for consistency at multiple points: 
1. Consistency of fake-quant state dict with loading a model from scratch
2. Consistency of state dict after convert with a model created using PTQ (the qconfig needs to be consistent: Both FQ and observer need to use the same type of quantization).